test/validation.html

```diff
@@ -0,0 +1,155 @@
+        checkValidity() {
+          return this.value === '2020-02-02T20:20';
+        }
+      }
+      customElements.define('vaadin-date-time-picker-2020', DateTimePicker2020Element);
+    });
+  </script>
+
+  <test-fixture id="custom-validation">
+    <template>
+      <vaadin-date-time-picker-2020></vaadin-date-time-picker-2020>
+    </template>
+  </test-fixture>
+
+  <script>
+    describe(`Validation`, () => {
+      let dateTimePicker;
+      let customField;
+
+      ['default', 'slotted'].forEach(set => {
+
+        describe(`Default (${set})`, () => {
+
+          beforeEach(() => {
+            if (set === 'default') {
+              dateTimePicker = fixture('default');
+              customField = dateTimePicker.$.customField;
+            } else if (set === 'slotted') {
+              dateTimePicker = fixture('slotted-inputs');
+              customField = dateTimePicker.$.customField;
+            }
+          });
+
+          afterEach(() => {
+            dateTimePicker.invalid = false;
+            dateTimePicker.errorMessage = '';
+          });
+
+          it('should be required', () => {
+            dateTimePicker.required = true;
+            expect(customField.required).to.be.true;
+          });
+
+          it('should not be required', () => {
+            expect(customField.required).to.be.false;
+          });
+
+          it('should validate correctly with required flag', done => {
+            dateTimePicker.name = 'foo';
+            dateTimePicker.required = true;
+
+            setTimeout(() => {
+              expect(dateTimePicker.validate()).to.equal(false);
+              expect(dateTimePicker.invalid).to.be.equal(true);
+
+              dateTimePicker.value = '2020-02-02T02:02:00';
+              expect(dateTimePicker.validate()).to.equal(true);
+              expect(dateTimePicker.invalid).to.be.equal(false);
+
+              done();
+            });
+          });
+
+          it('should validate min/max dates', () => {
+            dateTimePicker.min = '2020-02-02T02:00';
+            dateTimePicker.max = '2020-02-02T04:00';
+
+            // Set invalid value.
+            dateTimePicker.value = '2020-02-02T01:00';
+            expect(dateTimePicker.validate()).to.equal(false);
+            expect(dateTimePicker.invalid).to.be.equal(true);
+
+            dateTimePicker.value = '2020-02-02T03:00';
+            expect(dateTimePicker.validate()).to.equal(true);
+            expect(dateTimePicker.invalid).to.be.equal(false);
+          });
+
+          it('should be possible to force invalid status', () => {
+            dateTimePicker.invalid = true;
+            expect(customField.invalid).to.be.true;
+          });
+
+          it('should display the error-message when invalid', () => {
+            dateTimePicker.invalid = true;
+            dateTimePicker.errorMessage = 'Not cool';
+            expect(customField.errorMessage).to.equal(dateTimePicker.errorMessage);
+          });
+        });
```

Comment:
This test will pass even if the line `dateTimePicker.invalid = true;` is removed. This is only checking that the error-message property is being propagated to custom-field. It doesn't check that it is being displayed or not.

But I'm not sure if it even needs to be tested here. I think it might be enough just to test that both properties are propagated separately, since then it's the job of custom-field to actually display it (and that should also be tested in the custom-field repo).